.github/workflows/run-all-e2e-tests.yaml

```diff
@@ -0,0 +1,57 @@
+# Runs Azure tests when pull request opened, repopened or synchronized
+name: E2E Tests - Run all tests on PR
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [labeled, synchronize]
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  set-global-conditions:
```

Comment:
GHA workflows does not allow condition at workflow level that can be shared between all the jobs.
This hack allows setting complicated `if` condition on all the jobs and make it easy for dependent job to inherit condition.